validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockFactory.java

```diff
@@ -87,7 +87,7 @@ public BeaconBlock createUnsignedBlock(
     }
 
     // Collect attestations to include
-    final BeaconState blockSlotState = stateTransition.process_slots(previousState, newSlot);
+    final BeaconState blockSlotState = stateTransition.process_slots(blockPreState, newSlot);
```

Comment:
this would potentially call process_slots twice... i would have thought the point was that we don't need this at all?
`final BeaconState blockSlotState = blockPreState;` 
effectively